`network.informo.article` matrix event (#9)
Informo/specs: content/information-distribution/article.md

the node using [Matrix's content repository
module](https://matrix.org/docs/spec/client_server/r0.4.0.html#id112).

## Matrix event `network.informo.article`

This message event is meant to be sent by a source Matrix user. If the article

> is signed, it **must** be wrapped into a [Signed Matrix

Reviewer: "signed" shouldn't be capitalised.

Reviewer: This part should be left as is until #8 gets merged. Afterwards, two issues need to be addressed:

1. "If the article is signed" => articles are always signed
2. No mention who signs the event

Both can be solved by removing the paragraph, or rephrasing it, since, as it currently is phrased, #8 makes it redundant.

Author: IMO we should consider #8 merged as it will likely be merged before this PR (thus the link to the signing page)

Reviewer: Sure, I was suggesting to do it that way (then merge master into this branch) so we can have a full picture of the page's final state.

event](/information-distribution/signature/#signed-matrix-event).

If the sender is not a source user, the article **should** be ignored. If the
source registers itself afterwards, its previously sent articles **should**
become visible.

### Event data

|      Parameter      |   Type   | Req. |                                                                            Description                                                                            |
| ------------------- | -------- | :--: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `title`             | `string` |  x   | Article's headline                                                                                                                                                |

> | `href`              | `string` |      | Link to the article's original post (in case the article was sent to informo from an existing website)                                                            |

Reviewer: "Informo"'s "I" should be capitalised.

Reviewer: Also, `s/Link to/URL of/`

| `short_description` | `string` |      | Short description or introduction to the article.                                                                                                                 |
| `author`            | `string` |      | Full name of the article's author (when a single Informo source aggregates multiple writers).                                                                     |

> | `thumbnail`         | `string` |      | Preview image for the article. Must be a `mxc://` url.                                                                                                            |

Reviewer: Might be worth adding a link to the mxc specs (which would be https://matrix.org/docs/spec/client_server/r0.4.0.html#id112)

Reviewer: Also, please write "URL" in upper case since it's an acronym.

> | `date`              | `int`    |      | Timestamp in milliseconds when the article was published. If not provided clients should fall-back to the timestamp when the matrix event was sent.               |

Reviewer: Not sure of the phrasing here. I'd phrase it like this:

```suggestion
| `date`              | `int`    |      | Timestamp in milliseconds of the article's publication. If not provided clients should fall back to the Matrix event's creation timestamp.               |
```

| `content`           | `string` |  x   | Article HTML content. The HTML **must** be sanitized before being displayed in a client.                                                                          |
| `custom`            | `object` |      | Additional information for custom client implementations.                                                                                                         |

Additional information:

- Articles having a `date` field in the future **should** be ignored.
- The `date` field uses the same timestamps as in the Matrix protocol, i.e. milliseconds since epoch.

### Example

```json
{
    "content": {
        "algorithm": "ed25519"
        "sender_key": "IlRMeOPX2e0MurIyfWEucYBRVOEEUMrOHqn/8mLqMjA"
        "signature": "0a1df56f1c3ab5b1"
        "signed": {
            "title": "Lorem ipsum",
            "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
        }
    }
}

> ```

Reviewer: JSON isn't valid (lines don't end with `,`.

## Matrix event `network.informo.article`

This message event **must** be sent and signed by a source Matrix user. See

> [signed Matrix event](/information-distribution/signature/#signed-matrix-event)

Reviewer: Sorry to be picky here, but I think the rendered version of this page might not make the face that we refer to another page obvious enough. I'd rephrase that into "See the [signed Matrix event] page for more information...".

### Event data

|      Parameter      |   Type   | Req. |                                                                     Description                                                                     |
| ------------------- | -------- | :--: | --------------------------------------------------------------------------------------------------------------------------------------------------- |
| `title`             | `string` |  x   | Article's headline.                                                                                                                                 |
| `href`              | `string` |      | URL of the article's original post (in case the article was sent to Informo from an existing website).                                              |
| `short_description` | `string` |      | Short description or introduction to the article.                                                                                                   |
| `author`            | `string` |      | Full name of the article's author (when a single Informo source aggregates multiple writers).                                                       |

> | `thumbnail`         | `string` |      | Preview image for the article. Must be a [`mxc://` url](https://matrix.org/docs/spec/client_server/r0.4.0.html#id112).                                                                                              |

Reviewer: To forgot to spell URL in upper case here.

> | `date`              | `int`    |      | Timestamp in milliseconds when the article was published. If not provided clients should fall-back to the timestamp when the matrix event was sent. |

Reviewer: Same comment as from my previous review.
